### ISSUE TITLE
Fold in JS is using `+`

### ISSUE THREAD

**Issue by neumino** [rethinkdb/docs#1087]
There's a wild `+`. It should be a call to `add`
http://rethinkdb.com/api/javascript/fold/

**Comment by neumino** [comment]
Also the emit functions needs 3 arguments

**Comment by neumino** [comment]
One more thing, there's no example with `finalEmit`.

**Comment by neumino** [comment]
The function that takes 3 argument is the function passed to emit, not the first one pass to `fold`.
I believe what you want is:
```
r.table('even_things').fold(0, function(acc, row) {
    return acc.add(1);
}, {emit:
    function (acc, row, new_acc) {
        return r.branch(acc.mod(2).eq(0), [row], []);
    }
})
```

You also probably want to order these documents.

For the first example, you also probably want to map a field since you can't concatenate objects.

**Comment by chipotle** [comment]
The incorrect fold example was fixed as part of #1144.
